packages/mml-web/src/MMLClickTrigger.ts

```diff
@@ -54,6 +61,71 @@ export class MMLClickTrigger {
     }
   }
 
+  private handleTouchEnd(event: TouchEvent) {
+    if (Date.now() - touchTimestamp < touchThresholdMilliseconds) {
+      /* a short touch, i.e., a click */
+      if ((event.detail as any).element) {
+        // Avoid infinite loop of handling click events that originated from this trigger
+        return;
+      }
+      let x = 0;
+      let y = 0;
+      if (!document.pointerLockElement) {
+        let offsetX = touchX;
+        let offsetY = touchY;
+        let width = window.innerWidth;
+        let height = window.innerHeight;
+        if (this.clickTarget instanceof HTMLElement) {
+          width = this.clickTarget.offsetWidth;
+          height = this.clickTarget.offsetHeight;
+        }
+        if (event.target) {
+          /* get the equivalent of event.offset in a mouse event */
+          const bcr = (event.target as HTMLElement).getBoundingClientRect();
+          offsetX = offsetX - bcr.x;
+          offsetY = offsetY - bcr.y;
+        }
+        x = (offsetX / width) * 2 - 1;
+        y = -((offsetY / height) * 2 - 1);
+      }
+      this.raycaster.setFromCamera(new THREE.Vector2(x, y), this.scene.getCamera());
+      const intersections = this.raycaster.intersectObject(this.scene.getRootContainer(), true);
+      if (intersections.length > 0) {
+        for (const intersection of intersections) {
+          let obj: THREE.Object3D | null = intersection.object;
+          while (obj) {
+            /*
+               Ignore scene objects that have a transparent or wireframe material
+              */
+            if (this.isMaterialIgnored(obj)) {
+              break;
+            }
+
+            const mElement = MElement.getMElementFromObject(obj);
+            if (mElement && mElement.isClickable()) {
+              mElement.dispatchEvent(
+                new MouseEvent("click", {
+                  bubbles: true,
+                }),
+              );
+              return;
+            }
+            obj = obj.parent;
+          }
+        }
+      }
+    }
+  }
+
+  private handleTouchStart(event: TouchEvent) {
+    /* remember the x and y position of the touch, so that it can be used in touchEnd */
+    touchX = event.touches[0].clientX;
+    touchY = event.touches[0].clientY;
```

Comment:
Using the first-available touch to track position here (rather than using the position of the touch that has ended in `touchEnd`) has an unintentional side-effect.

If you hold a finger (A) on the screen, and then tap with a second finger (B), you will be "clicking" at point A rather than B which is where the actual tapping is happening.